strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/users/service/impl/OrientDbUserService.java

```diff
@@ -0,0 +1,208 @@
+import com.orientechnologies.orient.core.sql.OCommandSQL;
+import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
+
+/**
+ * @author sbespalov
+ */
+@Component
+@OrientDb
+public class OrientDbUserService extends CommonCrudService<UserEntry> implements UserEntryService
+{
+
+    @Inject
+    private SecurityTokenProvider tokenProvider;
+    
+    @Override
+    @CacheEvict(cacheNames = CacheName.User.AUTHENTICATIONS, key = "#p0")
+    public void deleteByUsername(String username)
+    {
+        Map<String, String> params = new HashMap<>();
+        params.put("username", username);
+
+        String sQuery = String.format("delete from %s where username = :username", UserEntry.class.getSimpleName());
+
+        OCommandSQL oQuery = new OCommandSQL(sQuery);
+        getDelegate().command(oQuery).execute(params);
+    }
+
+    @Override
+    public UserEntry findByUsername(String username)
+    {
+        Map<String, String> params = new HashMap<>();
+        params.put("username", username);
+
+        String sQuery = buildQuery(params);
+
+        OSQLSynchQuery<ODocument> oQuery = new OSQLSynchQuery<>(sQuery);
+        oQuery.setLimit(1);
+
+        List<UserEntry> resultList = getDelegate().command(oQuery).execute(params);
+        
+        return resultList.stream().findFirst().orElse(null);
+
+    }
+
+    @Override
+    public String generateSecurityToken(String username)
+        throws JoseException
+    {
+        final User user = findByUsername(username);
+
+        if (StringUtils.isEmpty(user.getSecurityTokenKey()))
+        {
+            return null;
+        }
+
+        final Map<String, String> claimMap = new HashMap<>();
+        claimMap.put(UserData.SECURITY_TOKEN_KEY, user.getSecurityTokenKey());
+
+        return tokenProvider.getToken(username, claimMap, null, null);
+    }
+
+    @Override
+    @CacheEvict(cacheNames = CacheName.User.AUTHENTICATIONS, key = "#p0.username")
+    public void updateAccountDetailsByUsername(User userToUpdate)
+    {
+        UserEntry user = findByUsername(userToUpdate.getUsername());
+        if (user == null) 
+        {
+            throw new UsernameNotFoundException(userToUpdate.getUsername());
+        }
+        
+        if (!StringUtils.isBlank(userToUpdate.getPassword()))
+        {
+            user.setPassword(userToUpdate.getPassword());
+        }
+
+        if (StringUtils.isNotBlank(userToUpdate.getSecurityTokenKey()))
+        {
+            user.setSecurityTokenKey(userToUpdate.getSecurityTokenKey());
+        }
+        
+        save(user);
+    }
+
+    @Override
+    public Users getUsers()
+    {
+        Optional<List<UserEntry>> allUsers = findAll();
+        if (allUsers.isPresent()) 
+        {
+            return new Users(allUsers.get().stream().map(u -> detach(u)).collect(Collectors.toSet()));
+        }
+        
+        return null;
+    }
+
+    @Override
+    public void revokeEveryone(String roleToRevoke)
+    {
+
```

Comment:
UnsupportedOperationException ?

Reply:
yeah, seems I've missed it, thanks

added the `UnsupportedOperationException`, and I will also create a follow up task for this

Reply:
implementation provided, thanks